### PR TITLE
actions: use `/bin/bash`

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
When there is a broken `bash` in `PATH`, some actions may fail as in Homebrew/homebrew-core#162819. (See [this job](https://github.com/Homebrew/homebrew-core/actions/runs/7916916851/job/21612022470?pr=162819).) Let's modify those actions to use `/bin/bash` instead of an arbitrary `bash` from `PATH`; `brew` requires and always runs on `/bin/bash` anyway.

### CODE REVIEW

**Comment by Bo98** [comment]
As a slight nit: `-euo pipefail` is more like what we had before but LGTM.

File: count-bottles/action.yml [hunk 1]
```diff
@@ -27,5 +27,5 @@ runs:
   steps:
     - run: ./count.sh '${{ inputs.working-directory }}' '${{ inputs.debug }}'
       working-directory: ${{ github.action_path }}
-      shell: bash
+      shell: /bin/bash -e {0}
```
Comment (MikeMcQuaid) [inline count-bottles/action.yml:30]:
```suggestion
      shell: /bin/bash -euo pipefail {0}
```

may want to consider that on all of these, assuming it doesn't break them.

**Review by MikeMcQuaid — APPROVED** [review]
Great move, one suggestion (that can be a later PR).

**Comment by ZhongRuoyu** [comment]
Thanks for the quick review @Bo98 @MikeMcQuaid!

**Pull request merged**